packages/office-ui-fabric-react/src/components/List/List.tsx

```diff
@@ -207,50 +213,65 @@ export class List extends BaseComponent<IListProps, IListState> implements IList
         // We have found the page. If the user provided a way to measure an individual item, we will try to scroll in just
         // the given item, otherwise we'll only bring the page into view
         if (measureItem) {
-          // Adjust for actual item position within page
-          const itemPositionWithinPage = index - itemIndex;
-          for (let itemIndexInPage = 0; itemIndexInPage < itemPositionWithinPage; ++itemIndexInPage) {
-            scrollTop += measureItem(itemIndex + itemIndexInPage);
-          }
-          const scrollBottom = scrollTop + measureItem(index);
 
           const scrollRect = _measureScrollRect(this._scrollElement);
           const scrollWindow = {
             top: this._scrollElement.scrollTop,
             bottom: this._scrollElement.scrollTop + scrollRect.height
           };
 
+          // Adjust for actual item position within page
+          const itemPositionWithinPage = index - itemIndex;
+          for (let itemIndexInPage = 0; itemIndexInPage < itemPositionWithinPage; ++itemIndexInPage) {
+            scrollTop += measureItem(itemIndex + itemIndexInPage);
+          }
+          const scrollBottom = scrollTop + measureItem(index);
+
           const itemIsFullyVisible = scrollTop >= scrollWindow.top && scrollBottom <= scrollWindow.bottom;
-          if (itemIsFullyVisible) {
-            // Item is already visible, do nothing.
+          // Item is already visible and scrollMode is auto, do nothing and return
+          if (itemIsFullyVisible && scrollToMode === ScrollToMode.auto) {
             return;
           }
 
           const itemIsPartiallyAbove = scrollTop < scrollWindow.top;
```

Comment:
I don't like how the new options are used with the old logic, can we restructure to make that clearer?

Reply:
Why calculate the `itemIsPartially...` values at all if your scrollToMode option is set?